index.js

```diff
@@ -1,37 +1,31 @@
 var markdown = require('markdown-it')
 var hljs = require('highlight.js')
-
-var extend = function (obj, source) {
-  var prop
-
-  for (prop in source) {
-    if (source.hasOwnProperty(prop)) {
-      obj[prop] = source[prop]
-    }
-  }
-
-  return obj
-}
+var loaderUtils = require('loader-utils')
 
 module.exports = function (source) {
   this.cacheable()
 
-  var opts = extend({
-    preset: 'default',
-    highlight: function (str, lang) {
-      if (lang && hljs.getLanguage(lang)) {
+  var opts = Object.assign(
+    {
+      preset: 'default',
+      highlight: function (str, lang) {
+        if (lang && hljs.getLanguage(lang)) {
+          try {
+            return hljs.highlight(lang, str).value
+          } catch (err) {}
+        }
+
         try {
-          return hljs.highlight(lang, str).value
+          return hljs.highlightAuto(str).value
```

Comment:
Why not use the language that was given to us? It feels wasteful to do language detection again.